connectors/slack/template/index.js

```diff
@@ -98,32 +106,29 @@ async function dispatch(ctx) {
 
   // Route all other events
   if (body && body.api_app_id && body.team_id) {
-    storage.ensureStorage(ctx);
-    await storage.ensureCache();
+    await storage.ensureCache(ctx);
     Sdk.debug('Event is for Team ID', body.api_app_id, body.team_id);
-    let tokens =
-      storage.teamToHandler[body.api_app_id][body.team_id] &&
-      Object.keys(storage.teamToHandler[body.api_app_id][body.team_id]);
-    if (!tokens) {
+    let handlers = storage.teamToHandler[body.api_app_id][body.team_id];
+    if (!handlers) {
       Sdk.debug('No handlers registered for this Team ID, not routing');
       return;
     }
 
-    const promises = tokens.map((t) =>
-      Superagent.post(storage.teamToHandler[body.api_app_id][body.team_id][t])
-        .set('Authorization', `Bearer ${t}`)
+    const promises = handlers.map((h) =>
+      Superagent.post(h)
+        .set('Authorization', `Bearer ${ctx.fusebit.functionAccessToken}`)
         .send(body.event)
         .ok((r) => {
           if (r.status >= 300) {
-            Sdk.debug('Handler error', storage.teamToHandler[body.api_app_id][body.team_id][t], r.body);
+            Sdk.debug('Handler error', h, r.body);
           }
           return true;
         })
     );
 
     try {
```

Comment:
Defensively, bring this up above the declaration of `promises`, to avoid any accidental async operations being added in between these two sections and causing exceptions to potentially be uncaught.

Reply:
Fixed